Sources/Jobs/Job+Traversable.swift

```diff
@@ -0,0 +1,38 @@
+import Apodini
+
+/// Extension that defines methods that traverses properties of `Job`s.`
+extension Job {
+    /// Activates all properties of `Job`s at once and always keeps them activated.
+    func activate() -> Self {
+        var selfCopy = self
+
+        apply({ (activatable: inout Activatable) in
+            activatable.activate()
+        }, to: &selfCopy)
+
+        return selfCopy
+    }
+    
+    /// Checks if only valid property wrappers are used with `Job`s.
+    func checkPropertyWrapper() throws {
+        try execute({ (_ :Environment<EnvironmentValues, Connection>) in
+            throw JobErrors.requestPropertyWrapper
+        }, on: self)
+        
+        try execute({ (_: RequestBasedPropertyWrapper) in
+            throw JobErrors.requestPropertyWrapper
+        }, on: self)
+    }
+    
+    /// Subscribes to all `ObservedObject`s.
+    func subscribe(configuration: JobConfiguration) {
+        execute({ (observedObject: AnyObservedObject) in
+            observedObject.valueDidChange = {
+                observedObject.setChanged(to: true)
+                // Executes the `Job` on its own event loop
+                _ = configuration.scheduled?.futureResult.hop(to: configuration.eventLoop)
+                observedObject.setChanged(to: false)
```

Comment:
I don't think this logic works as intended. The `observedObject.setChanged` should be called **after** the future has completed, shouldn't it?

Reply:
Yes, you're right this makes it safer

Reply:
Fixed with `c3c4523`

Reply:
Instead of "hopping" I'm executing it now directly on the eventloop